consensus/XDPoS/XDPoS.go

```diff
@@ -346,86 +265,44 @@ func (x *XDPoS) GetMasternodesByNumber(chain consensus.ChainReader, blockNumber
 		log.Error("[GetMasternodesByNumber] Unable to find block", "Num", blockNumber)
 		return []common.Address{}
 	}
-	switch x.config.BlockConsensusVersion(big.NewInt(int64(blockNumber)), blockHeader.Extra, ExtraFieldCheck) {
-	case params.ConsensusEngineVersion2:
-		return x.EngineV2.GetMasternodes(chain, blockHeader)
-	default: // Default "v1"
-		return x.EngineV1.GetMasternodes(chain, blockHeader)
-	}
+	return x.EngineV2.GetMasternodes(chain, blockHeader)
 }
 
 func (x *XDPoS) YourTurn(chain consensus.ChainReader, parent *types.Header, signer common.Address) (bool, error) {
-	switch x.config.BlockConsensusVersion(big.NewInt(parent.Number.Int64()+1), nil, SkipExtraFieldCheck) {
-	case params.ConsensusEngineVersion2:
-		return x.EngineV2.YourTurn(chain, parent, signer)
-	default: // Default "v1"
-		return x.EngineV1.YourTurn(chain, parent, signer)
-	}
+	return x.EngineV2.YourTurn(chain, parent, signer)
 }
 
 func (x *XDPoS) GetValidator(creator common.Address, chain consensus.ChainReader, header *types.Header) (common.Address, error) {
```

Comment:
do we still need those v1 specific methods? probably can be deleted all together right?

Reply:
Just checked, to fully clean up this function `GetValidator()` will need `eth/fetcher/fetcher.go`, `eth/backend.go` changes too. I prefer keep it now. Too much changes at same time.

Reply:
Same as other V1 functions. So just keep the function will `nil` is quick way to have light clean up